analyzers/tests/SonarAnalyzer.UnitTest/Extensions/BlockSyntaxExtensionsTests.cs

```diff
@@ -0,0 +1,316 @@
+        }
+
+        [TestMethod]
+        public void IsEmpty_BlockMethodWithConditionalCompilation_WithEmptyConditional_On()
+        {
+            var block = @"
+{
+#if SomeCondition
+#endif
+}
+";
+            var declaration = MethodDeclarationForBlock(block);
+            declaration.Body.Should().BeEmpty(treatConditionalCompilationAsContent: true);
+        }
+
+        [DataTestMethod]
+        [DataRow(true, true, false)]
+        [DataRow(false, true, false)] // isEmpty should be true here, because the content of the conditional is just a comment and a comment should not be treated as content.
+        [DataRow(true, false, true)]
+        [DataRow(false, false, true)]
+        public void IsEmpty_BlockMethodCombinations_CommentInConditional(bool treatCommentsAsContent, bool treatConditionalCompilationAsContent, bool isEmpty)
+        {
+            var block = @"
+{
+#if SomeCondition
+    // Some comment
+#endif
+}
+";
+            var declaration = MethodDeclarationForBlock(block);
+            if (isEmpty)
+            {
+                declaration.Body.Should().BeEmpty(treatCommentsAsContent, treatConditionalCompilationAsContent);
+            }
+            else
+            {
+                declaration.Body.Should().BeNotEmpty(treatCommentsAsContent, treatConditionalCompilationAsContent);
+            }
+        }
+
+        [TestMethod]
+        public void IsEmpty_TrailingTriviaOnOpenBrace_Comment()
+        {
+            var block = BlockWithoutTrivia();
+            block = block.WithOpenBraceToken(Token(SyntaxKind.OpenBraceToken).WithTrailingTrivia(ParseTrailingTrivia("// Comment")));
+            block.Should().BeNotEmpty(treatCommentsAsContent: true);
+        }
+
+        [TestMethod]
+        public void IsEmpty_LeadingTriviaOnCloseBrace_Comment()
+        {
+            var block = BlockWithoutTrivia();
+            block = block.WithCloseBraceToken(Token(SyntaxKind.CloseBraceToken).WithLeadingTrivia(ParseLeadingTrivia("// Comment")));
+            block.Should().BeNotEmpty(treatCommentsAsContent: true);
+        }
+
+        [TestMethod]
+        public void IsEmpty_TrailingTriviaOnOpenBrace_ConditionalCompilation()
+        {
+            var block = BlockWithoutTrivia();
+            block = block.WithOpenBraceToken(Token(SyntaxKind.OpenBraceToken).WithTrailingTrivia(ParseTrailingTrivia("#if Something\r\n var i = 1; \r\n #endif")));
+            block.Should().BeNotEmpty(treatConditionalCompilationAsContent: true);
+        }
+
+        [TestMethod]
+        public void IsEmpty_LeadingTriviaOnCloseBrace_ConditionalCompilation()
+        {
+            var block = BlockWithoutTrivia();
+            block = block.WithCloseBraceToken(Token(SyntaxKind.CloseBraceToken).WithLeadingTrivia(ParseLeadingTrivia("#if Something\r\n var i = 1; \r\n #endif")));
+            block.Should().BeNotEmpty(treatConditionalCompilationAsContent: true);
+        }
```

Comment:
I would refactor the tests to be like:
```
public void IsEmpty_TrailingTriviaOnOpenBrace_Comment() =>
            CreateBlock(CreateTriviaListWithComment(), default, default, default).Should().BeNotEmpty(treatCommentsAsContent: true);
```
by introducing:
```
        private static SyntaxTriviaList CreateTriviaListWithComment() =>
            new(SyntaxTrivia(SyntaxKind.SingleLineCommentTrivia, " "));

        private static BlockSyntax CreateBlock(SyntaxTriviaList beforeOpen, SyntaxTriviaList afterOpen, SyntaxTriviaList beforeClose, SyntaxTriviaList afterClose) =>
            Block(Token(beforeOpen, SyntaxKind.OpenBraceToken, afterOpen), default, Token(beforeClose, SyntaxKind.CloseBraceToken, afterClose));
```

And then we can remove `BlockWithoutTrivia()`.